statsmodels/tsa/statespace/mlemodel.py

```diff
@@ -0,0 +1,797 @@
+
+    Attributes
+    ----------
+    aic : float
+        Akaike Information Criterion
+    bic : float
+        Bayes Information Criterion
+    bse : array
+        The standard errors of the parameters. Computed using the numerical
+        Hessian.
+    cov_params : array
+        The variance / covariance matrix. Computed using the numerical Hessian.
+    hqic : array
+        Hannan-Quinn Information Criterion
+    llf : float
+        The value of the log-likelihood function evaluated at `params`.
+    model : Model instance
+        A reference to the model that was fit.
+    nobs : float
+        The number of observations used to fit the model.
+    params : array
+        The parameters of the model.
+    pvalues : array
+        The p-values associated with the z-statistics of the coefficients.
+        Note that the coefficients are assumed to have a Normal distribution.
+    scale : float
+        This is currently set to 1.0 and not used by the model or its results.
+    sigma2 : float
+        The variance of the residuals.
+    zvalues : array
+        The z-statistics for the coefficients.
+
+    Methods
+    -------
+    conf_int
+    f_test
+    fittedvalues
+    forecast
+    load
+    predict
+    remove_data
+    resid
+    save
+    summary
+    t_test
+    wald_test
+    """
+    def __init__(self, model, *args, **kwargs):
+        self.data = model.data
+
+        # Save the model output
+        self._endog_names = model.endog_names
+        self._exog_names = model.endog_names
+        self._params = model.params
+        self._params_names = model.params_names
+        self._model_names = model.model_names
+        self._model_latex_names = model.model_latex_names
+
+        # Associate the names with the true parameters
+        params = pd.Series(self._params, index=self._params_names)
+
+        # Initialize the Statsmodels model base
+        tsbase.TimeSeriesModelResults.__init__(self, model, params,
+                                               normalized_cov_params=None,
+                                               scale=1., *args, **kwargs)
+
+        # Initialize the statespace representation
+        super(MLEResults, self).__init__(model)
+
+        # Setup the cache
+        self._cache = resettable_cache()
+
+    @cache_readonly
+    def aic(self):
+        # return -2*self.llf + 2*self.params.shape[0]
+        return aic(self.llf, self.nobs, self.params.shape[0])
+
+    @cache_readonly
+    def bic(self):
+        # return -2*self.llf + self.params.shape[0]*np.log(self.nobs)
+        return bic(self.llf, self.nobs, self.params.shape[0])
+
+    @cache_readonly
+    def bse(self):
+        return np.sqrt(np.diagonal(self.cov_params))
+
+    @cache_readonly
+    def cov_params(self):
+        hessian = self.model.hessian(
+            self._params, set_params=False, transformed=True,
+            initial_state=self.initial_state,
+            initial_state_cov=self.initial_state_cov
+        )
```

Comment:
> If that is the wrong one (because of transform params), then we should avoid calculating it. master has an option for it, but it might not be available until a rebase of this branch.

Yes, I think that it will not be the correct one. For now, I will set `skip_hessian=True`.

> Most likely the Hessian calculation should move into fit as in other models, but we might need to provide other cov_type changes at the same time. (leave it for another PR)

I'm not very familiar with the `fit()` method in LikelihoodBase and the way it interacts with `Optimizer`, but I'm open to changing how this works. This would probably be good to do in a future PR in combination with adding the OPG method for `cov_params` along with the `likelihood_obs`, etc.